File: common-library/src/main/java/com/kingkong/common_library/utils/glide/BitmapProcessNode.java
package com.kingkong.common_library.utils.glide;

import android.graphics.Bitmap;
/**
 * 位图处理点
 */
public abstract class BitmapProcessNode {
    private BitmapProcessNode nextNode;

    public BitmapProcessNode() {}

    BitmapProcessNode connect(BitmapProcessNode nextNode) {
        this.nextNode = nextNode;
        return nextNode;
    }

    public final Bitmap processBitmap(Bitmap bitmap) {
        if(bitmap != null) {
            bitmap = onProcessBitmap(bitmap);
        }
        if(nextNode != null) {
            return nextNode.processBitmap(bitmap);
        }
        return bitmap;
    }

    protected abstract Bitmap onProcessBitmap(Bitmap bitmap);
}

File: common-library/src/main/java/com/kingkong/common_library/utils/glide/BitmapProcessNode.kt
package com.kingkong.common_library.utils.glide

import android.graphics.Bitmap

abstract class BitmapProcessNode {

    private var nextNode: BitmapProcessNode? = null

    constructor()

    open fun connect(nextNode: BitmapProcessNode?): BitmapProcessNode? {
        this.nextNode = nextNode
        return nextNode
    }

    fun processBitmap(bitmap: Bitmap?): Bitmap? {
        var bitmap = bitmap
        if (bitmap != null) {
            bitmap = onProcessBitmap(bitmap)
        }
        return nextNode?.processBitmap(bitmap) ?: bitmap
    }

    protected abstract fun onProcessBitmap(bitmap: Bitmap?): Bitmap
}